addons/mail/static/src/core/common/quick_reaction_menu.js

```diff
@@ -34,7 +34,21 @@ export class QuickReactionMenu extends Component {
                 popoverClass: "o-mail-QuickReactionMenu-pickerPopover",
             }
         );
-        this.dropdown = useState(useDropdownState());
+        this.dropdown = useState(
+            useDropdownState({
+                onClose: () => {
+                    const currentThread = this.env.getCurrentThread();
+                    if (!currentThread || currentThread.notEq(this.props.message.thread)) {
+                        return;
+                    }
+                    if (currentThread.messageInEdition) {
+                        currentThread.messageInEdition.composer.autofocus++;
+                    } else {
+                        currentThread.composer.autofocus++;
+                    }
+                },
+            })
+        );
```

Comment:
I thought that replying to a message in Inbox and using quick reaction then ESC would not refocus the composer. But apparently this works (already before this PR)! 

I'm wondering if this isn't `useActiveElement()` of popover that re-focuses the old focused element? Haven't investigated, but it just works anyway.